backend/src/org/commcare/xml/SuiteParser.java

```diff
@@ -75,19 +77,25 @@ public Suite parse() throws InvalidStructureException, IOException, XmlPullParse
                         //resource def
                         parser.nextTag();
                         Resource r = new ResourceParser(parser, maximumResourceAuthority).parse();
-                        table.addResource(r, table.getInstallers().getLocaleFileInstaller(localeKey), resourceGuid);
+                        if(!skipResources) {
+                            table.addResource(r, table.getInstallers().getLocaleFileInstaller(localeKey), resourceGuid);
+                        }
                     } else if (parser.getName().toLowerCase().equals("media")) {
                         String path = parser.getAttributeValue(null, "path");
                         //Can be an arbitrary number of resources inside of a media block.
                         while (this.nextTagInBlock("media")) {
                             Resource r = new ResourceParser(parser, maximumResourceAuthority).parse();
-                            table.addResource(r, table.getInstallers().getMediaInstaller(path), resourceGuid);
+                            if(!skipResources) {
+                                table.addResource(r, table.getInstallers().getMediaInstaller(path), resourceGuid);
+                            }
                         }
                     } else if (parser.getName().toLowerCase().equals("xform")) {
                         //skip xform stuff for now
                         parser.nextTag();
                         Resource r = new ResourceParser(parser, maximumResourceAuthority).parse();
```

Comment:
We're stream processing, so we can't in this case, we need to consume the full document in order to move on, and can't skip stuff.